### PR TITLE
Load sidebar filters and results asynchronously

### PR DESCRIPTION
**Pull request by matt-downs**
Changes in this PR: 
- Added some truly empty header and footer templates
- All product filters from the sidebar are moved to a different body template
- The filters are then loaded after the initial pageload by combining the current page URL (including any applied filters) with the `templatebody=filters`, `templatehead=none` and `templatefoot=none` params
- The response from this request is then injected into the `#filters` element

This more of a proof of concept - feedback and changes are welcome!

### CODE REVIEW

**Comment by craigblunden** [comment]
I really like the idea, but why not just use built in B@SE to reduce JS bloat? 

https://developers.neto.com.au/documentation/neto-designer-documentation/b-se-tag-library/function-tags/templating/load_ajax_template/

```
[%load_ajax_template id:'_filters' type:'content' template:'filters' preload:'0' /%]
[%site_value id:'footer_javascript'%]
    <script type="text/javascript" language="javascript">
        $(function(){
            $.load_ajax_template('_filters');
        })
    </script>
[%/site_value%]
```

**Comment by matt-downs** [comment]
Another optimisation to this would be to preload the fetch request in the head of the templates like so: 
```
<link rel="preload" as="fetch" href="/?rf=kw&kw=&templatebody=filters&templatehead=none&templatefoot=none">
```

The only issue with this is that I can't find a tag or function that can be used to grab the current full URL including query strings. 

We could manually build the URL using multiple `[@form:val@]` tags but this could easily break if our filter query strings were ever changed.

File: src/templates/cms/includes/sidebar.template.html [hunk 1]
```diff
@@ -1,145 +1,48 @@
...
+			<script>
+			$(document).ready(function() {
+				var seperator = window.location.href.indexOf("?") > -1 ? "&" : "?";
+				var url = window.location.href + seperator + "templatebody=filters&templatehead=none&templatefoot=none"
+
+				$.get(url, function(data) {
+					$('#filters').html(data);
+
+					// Init price slider
+					var dataset = document.querySelector('#price-filter').dataset
```
Comment (matt-downs) [inline src/templates/cms/includes/sidebar.template.html]: Browser support for `dataset` is also questionable... 

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

Comment (craigblunden) [reply src/templates/cms/includes/sidebar.template.html]: IE10 is disabled by default for TSL 1.2 (whatever that means?)
screen shot 2018-03-20 at 11 40 01 am 1

**Comment by markjaniczak** [comment]
How feasible would it be to load the category body asynchronously as you apply filters?

**Comment by craigblunden** [comment]
I tried it once, I don't think it works because you can't reapply all the
same filters that you've set from memory

**Comment by markjaniczak** [comment]
I think @matt-downs pulled it off by letting Neto generate the URLs and refreshing the filters with the body

**Comment by matt-downs** [comment]
Will update the PR tomorrow 😁

**Comment by philcon93** [comment]
Closing because of inactivity

**Pull request closed without merging**